File: app/src/main/java/com/peakmain/basicui/activity/home/loading/LoadingActivity.java
package com.peakmain.basicui.activity.home.loading;

import android.os.Handler;
import android.os.Looper;
import android.support.v7.widget.DividerItemDecoration;
import android.support.v7.widget.RecyclerView;

import com.peakmain.basicui.R;
import com.peakmain.basicui.adapter.BaseRecyclerStringAdapter;
import com.peakmain.basicui.base.BaseActivity;
import com.peakmain.basicui.utils.ActivityUtil;
import com.peakmain.ui.loading.CircleLoadingView;

import java.util.ArrayList;
import java.util.List;

/**
 * author ：Peakmain
 * createTime：2020/3/17
 * mail:2726449200@qq.com
 * describe：
 */
public class LoadingActivity extends BaseActivity {
    private RecyclerView mRecyclerView;
    private List<String> mLoadingBean;
    private BaseRecyclerStringAdapter mAdapter;
    private CircleLoadingView mLoadingView;

    private Handler mHandler = new Handler(Looper.getMainLooper());

    @Override
    protected int getLayoutId() {
        return R.layout.basic_recycler_view;
    }

    @Override
    protected void initView() {
        mRecyclerView = findViewById(R.id.recycler_view);
        mNavigationBuilder.setTitleText("加载loading").create();
    }

    @Override
    protected void initData() {
        mLoadingBean = new ArrayList<>();
        mLoadingBean.add("花束加载loading");
        mLoadingBean.add("视察动画的loading");
        mAdapter = new BaseRecyclerStringAdapter(this, mLoadingBean);
        mRecyclerView.addItemDecoration(new DividerItemDecoration(this, DividerItemDecoration.VERTICAL));
        mRecyclerView.setAdapter(mAdapter);
        mAdapter.setOnItemClickListener(position -> {
            switch (position) {
                case 0:
                    mLoadingView = new CircleLoadingView(this);
                    mLoadingView.show();
                    mHandler.postDelayed(() -> mLoadingView.hide(), 2000);
                    break;
                case 1:
                    ActivityUtil.gotoActivity(this,InspectLoadingActvivity.class);
                    break;
                case 2:
                    break;
                default:
                    break;
            }
        });
    }
}

File: app/src/main/java/com/peakmain/basicui/activity/home/loading/LoadingActivity.kt
package com.peakmain.basicui.activity.home.loading

import android.os.Handler
import android.os.Looper
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.RecyclerView
import com.peakmain.basicui.R
import com.peakmain.basicui.adapter.BaseRecyclerStringAdapter
import com.peakmain.basicui.base.BaseActivity
import com.peakmain.basicui.utils.ActivityUtil
import com.peakmain.ui.loading.CircleLoadingView
import com.peakmain.ui.recyclerview.listener.OnItemClickListener
import java.util.*

/**
 * author ：Peakmain
 * createTime：2020/3/17
 * mail:2726449200@qq.com
 * describe：
 */
class LoadingActivity : BaseActivity() {
    private var mRecyclerView: RecyclerView? = null
    private lateinit var mLoadingBean: MutableList<String>
    private var mAdapter: BaseRecyclerStringAdapter? = null
    private var mLoadingView: CircleLoadingView? = null
    private val mHandler = Handler(Looper.getMainLooper())
    override fun getLayoutId(): Int {
        return R.layout.basic_linear_recycler_view
    }

    override fun initView() {
        mRecyclerView = findViewById(R.id.recycler_view)
        mNavigationBuilder!!.setTitleText("加载loading").create()
    }

    override fun initData() {
        mLoadingBean = ArrayList()
        mLoadingBean.add("花束加载loading")
        mLoadingBean.add("仿老版58同城加载loading")
        mAdapter = BaseRecyclerStringAdapter(this, mLoadingBean)
        mRecyclerView!!.addItemDecoration(DividerItemDecoration(this, DividerItemDecoration.VERTICAL))
        mRecyclerView!!.adapter = mAdapter
        mAdapter!!.setOnItemClickListener(object : OnItemClickListener {
            override fun onItemClick(position: Int) {
                when (position) {
                    0 -> {
                        mLoadingView = CircleLoadingView(this@LoadingActivity)
                        mLoadingView!!.show()
                        mHandler.postDelayed({ mLoadingView!!.hide() }, 2000)
                    }
                    1 -> ActivityUtil.gotoActivity(this@LoadingActivity, ShapeLoadingActivity::class.java)
                    else -> {
                    }
                }
            }
        })
    }
}